### PR TITLE
An example use of eds_bias method is added

### PR DESCRIPTION
**Pull request by geemi725**
Addresses #98 #109

### CODE REVIEW

File: examples/EDS_tutorial.ipynb [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    "\n",
+    "    graph =htf.graph_builder(NN,output_forces=True)\n",
+    "    com = tf.reduce_mean(graph.positions[:, :2], 0) #calculate center of mass\n",
+    "    rs = graph.safe_norm(tf.math.subtract(graph.positions[:, :2], com), axis=1) #calculate distance of each atom from center of mass\n",
+    "    real_cv = tf.reduce_mean(rs) #calculate the average distance from center of mass. This is the collective variable (CV)\n",
+    "    cv_run=graph.running_mean(tensor=real_cv,name='cv_run') #calculates the running mean of the CV\n",
+    "    eds_alpha = htf.eds_bias(real_cv, set_point=set_pt, period=300,learning_rate=5.0) #Calculates the EDS alpha value every 300 steps. See htf/utils.py for more information\n",
+    "    eds_energy = eds_alpha * real_cv #computes EDS energy\n",
+    "    eds_forces = graph.compute_forces(eds_energy) #computes EDS forces\n",
+    "    tf.summary.scalar('collective variable',cv_run) #write a scalar summary of CV for tensorboard\n",
```
Comment (hgandhi2411) [inline examples/EDS_tutorial.ipynb]: @geemi725 can you have the comments be on a separate line, preferably above the line of code that you are explaining.

File: examples/EDS_tutorial.ipynb [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+   "outputs": [],
+   "source": [
+    "#### Build training graph ####\n",
+    "#saved as com_build.py in this example\n",
+    "import hoomd.htf as htf\n",
+    "import sys\n",
+    "import tensorflow as tf\n",
+    "\n",
+    "def make_eds_graph(N, NN, directory,set_pt):\n",
+    "    #N= Number of atoms in the system, NN=Number of nearest neighbors, directory=directory to save the results, set_pt=set point in EDS method\n",
```
Comment (hgandhi2411) [inline examples/EDS_tutorial.ipynb]: Very long comment,can be possibly broken up into smaller comments.

File: examples/EDS_tutorial.ipynb [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "### Hoomd-Sim code ###\n",
+    "import hoomd\n",
+    "import hoomd.md\n",
+    "import hoomd.htf\n",
+    "import tensorflow as tf\n",
+    "import sys\n",
+    "import com_build #import the graph building script\n",
```
Comment (hgandhi2411) [inline examples/EDS_tutorial.ipynb]: In a notebook you don't need to save as a file. You can directly use the function in the previous cell.

File: examples/EDS_tutorial.ipynb (hunk 1 above)
Comment (geemi725) [reply examples/EDS_tutorial.ipynb]: Done!

File: examples/EDS_tutorial.ipynb (hunk 2 above)
Comment (geemi725) [reply examples/EDS_tutorial.ipynb]: Comment broken into separate lines.

**Comment by RainierBarrett** [comment]
I have some things to address:

1. It's bad style to have `import` statements at the top of every cell in the notebook. You only need to import once per notebook.
2. The sections should not be code cells. We should use Markdown for this.
3. I have an error when I try to actually run these cells. First cell with the `def make_eds_graph` runs fine, but when I try to run the next code cell, I get the following:

> INFO:tensorflow:Summary name collective variable is illegal; using collective_variable instead.
> INFO:tensorflow:Summary name collective variable is illegal; using collective_variable instead.
> WARNING: You did not provide a virial for /scratch/rbarret8/eds_tutorial, so per particle virials will not be correct
> Note: Backed-up /scratch/rbarret8/eds_tutorial previous model to /scratch/rbarret8/eds_tutorial/previous_model_0
> notice(2): Started TF Session Manager.
> notice(2): Sending exit signal.
> notice(2): Shutting down TF Manually.
> notice(2): TF Queue is waiting, sending None
> Usage: ipykernel_launcher.py [options]
> 
> ipykernel_launcher.py: error: no such option: -f
> Exception in thread Thread-5:
> Traceback (most recent call last):
>   File "/home/rbarret8/.conda/envs/general_structs/lib/python3.6/threading.py", line 916, in _bootstrap_inner
>     self.run()
>   File "/home/rbarret8/.conda/envs/general_structs/lib/python3.6/threading.py", line 864, in run
>     self._target(*self._args, **self._kwargs)
>   File "/home/rbarret8/hoomd-blue/build/hoomd/htf/tfmanager.py", line 21, in main
>     write_tensorboard=write_tensorboard, **tfm_args)
> TypeError: type object argument after ** must be a mapping, not NoneType
> 
> ERROR:root:Internal Python error in the inspect module.
> Below is the traceback from this internal error.
> 
> Traceback (most recent call last):
>   File "/home/rbarret8/.conda/envs/general_structs/lib/python3.6/optparse.py", line 1387, in parse_args
>     stop = self._process_args(largs, rargs, values)
>   File "/home/rbarret8/.conda/envs/general_structs/lib/python3.6/optparse.py", line 1431, in _process_args
>     self._process_short_opts(rargs, values)
>   File "/home/rbarret8/.conda/envs/general_structs/lib/python3.6/optparse.py", line 1513, in _process_short_opts
>     raise BadOptionError(opt)
> optparse.BadOptionError: no such option: -f
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/home/rbarret8/.conda/envs/general_structs/lib/python3.6/site-packages/IPython/core/interactiveshell.py", line 3331, in run_code
>     exec(code_obj, self.user_global_ns, self.user_ns)
>   File "<ipython-input-5-f105871a4c0d>", line 14, in <module>
>     hoomd.context.initialize()
>   File "/home/rbarret8/hoomd-blue/build/hoomd/context.py", line 228, in initialize
>     hoomd.option._parse_command_line(args);
>   File "/home/rbarret8/hoomd-blue/build/hoomd/option.py", line 96, in _parse_command_line
>     (cmd_options, args) = parser.parse_args(args=input_args);
>   File "/home/rbarret8/.conda/envs/general_structs/lib/python3.6/optparse.py", line 1389, in parse_args
>     self.error(str(err))
>   File "/home/rbarret8/.conda/envs/general_structs/lib/python3.6/optparse.py", line 1569, in error
>     self.exit(2, "%s: error: %s\n" % (self.get_prog_name(), msg))
>   File "/home/rbarret8/.conda/envs/general_structs/lib/python3.6/optparse.py", line 1559, in exit
>     sys.exit(status)
> SystemExit: 2
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/home/rbarret8/.conda/envs/general_structs/lib/python3.6/site-packages/IPython/core/ultratb.py", line 1151, in get_records
>     return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)
>   File "/home/rbarret8/.conda/envs/general_structs/lib/python3.6/site-packages/IPython/core/ultratb.py", line 319, in wrapped
>     return f(*args, **kwargs)
>   File "/home/rbarret8/.conda/envs/general_structs/lib/python3.6/site-packages/IPython/core/ultratb.py", line 353, in _fixed_getinnerframes
>     records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))
>   File "/home/rbarret8/.conda/envs/general_structs/lib/python3.6/inspect.py", line 1490, in getinnerframes
>     frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)
> AttributeError: 'tuple' object has no attribute 'tb_frame'
> An exception has occurred, use %tb to see the full traceback.
> 
> ---------------------------------------------------------------------------
> 
> During handling of the above exception, another exception occurred:
> 
> SystemExit: 2

I'm not sure if this is an issue with my setup of the jupyter notebook on BH or something else. @hgandhi2411 , @whitead have you guys run into this before?

**Comment by hgandhi2411** [comment]
@geemi725 I'm unable to run your notebook. I believe it may be because our tensorflow versions are different. This is the error I get: 
```
Traceback (most recent call last):
  File "/gpfs/fs1/sfw2/python3/3.7.1/lib/python3.7/site-packages/IPython/core/interactiveshell.py", line 3267, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-1-2834c828469a>", line 1, in <module>
    import tensorflow as tf
  File "/home/hgandhi/.local/lib/python3.7/site-packages/tensorflow/__init__.py", line 99, in <module>
    from tensorflow_core import *
  File "/home/hgandhi/.local/lib/python3.7/site-packages/tensorflow_core/__init__.py", line 28, in <module>
    from tensorflow.python import pywrap_tensorflow  # pylint: disable=unused-import
  File "/home/hgandhi/.local/lib/python3.7/site-packages/tensorflow/__init__.py", line 50, in __getattr__
    module = self._load()
  File "/home/hgandhi/.local/lib/python3.7/site-packages/tensorflow/__init__.py", line 44, in _load
    module = _importlib.import_module(self.__name__)
  File "/software/anaconda3/2018.12b/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/home/hgandhi/.local/lib/python3.7/site-packages/tensorflow_core/python/__init__.py", line 73, in <module>
    from tensorflow.python.ops.standard_ops import *
  File "/home/hgandhi/.local/lib/python3.7/site-packages/tensorflow_core/python/ops/standard_ops.py", line 25, in <module>
    from tensorflow.python import autograph
  File "/home/hgandhi/.local/lib/python3.7/site-packages/tensorflow_core/python/autograph/__init__.py", line 35, in <module>
    from tensorflow.python.autograph import operators
  File "/home/hgandhi/.local/lib/python3.7/site-packages/tensorflow_core/python/autograph/__init__.py", line 37, in <module>
    from tensorflow.python.autograph.core.converter import ConversionOptions
  File "/home/hgandhi/.local/lib/python3.7/site-packages/tensorflow_core/python/autograph/core/converter.py", line 71, in <module>
    from tensorflow.python.autograph.pyct import cfg
  File "/home/hgandhi/.local/lib/python3.7/site-packages/tensorflow_core/python/autograph/pyct/cfg.py", line 41, in <module>
    from tensorflow.python.autograph.pyct import compiler
  File "/home/hgandhi/.local/lib/python3.7/site-packages/tensorflow_core/python/autograph/pyct/compiler.py", line 36, in <module>
    from tensorflow.python.autograph.pyct import origin_info
  File "/home/hgandhi/.local/lib/python3.7/site-packages/tensorflow_core/python/autograph/pyct/origin_info.py", line 31, in <module>
    from tensorflow.python.autograph.pyct import pretty_printer
  File "/home/hgandhi/.local/lib/python3.7/site-packages/tensorflow_core/python/autograph/pyct/pretty_printer.py", line 24, in <module>
    import termcolor
ModuleNotFoundError: No module named 'termcolor'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/gpfs/fs1/sfw2/python3/3.7.1/lib/python3.7/site-packages/IPython/core/interactiveshell.py", line 2018, in showtraceback
    stb = value._render_traceback_()
AttributeError: 'ModuleNotFoundError' object has no attribute '_render_traceback_'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/gpfs/fs1/sfw2/python3/3.7.1/lib/python3.7/site-packages/IPython/core/ultratb.py", line 1095, in get_records
    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)
  File "/gpfs/fs1/sfw2/python3/3.7.1/lib/python3.7/site-packages/IPython/core/ultratb.py", line 313, in wrapped
    return f(*args, **kwargs)
  File "/gpfs/fs1/sfw2/python3/3.7.1/lib/python3.7/site-packages/IPython/core/ultratb.py", line 347, in _fixed_getinnerframes
    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))
  File "/software/anaconda3/2018.12b/lib/python3.7/inspect.py", line 1500, in getinnerframes
    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)
  File "/software/anaconda3/2018.12b/lib/python3.7/inspect.py", line 1458, in getframeinfo
    filename = getsourcefile(frame) or getfile(frame)
  File "/software/anaconda3/2018.12b/lib/python3.7/inspect.py", line 696, in getsourcefile
    if getattr(getmodule(object, filename), '__loader__', None) is not None:
  File "/software/anaconda3/2018.12b/lib/python3.7/inspect.py", line 733, in getmodule
    if ismodule(module) and hasattr(module, '__file__'):
  File "/home/hgandhi/.local/lib/python3.7/site-packages/tensorflow/__init__.py", line 50, in __getattr__
    module = self._load()
  File "/home/hgandhi/.local/lib/python3.7/site-packages/tensorflow/__init__.py", line 44, in _load
    module = _importlib.import_module(self.__name__)
  File "/software/anaconda3/2018.12b/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 953, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/hgandhi/.local/lib/python3.7/site-packages/tensorflow_core/__init__.py", line 34, in <module>
    from tensorflow._api.v1 import autograph
  File "/home/hgandhi/.local/lib/python3.7/site-packages/tensorflow_core/_api/v1/autograph/__init__.py", line 22, in <module>
    from tensorflow._api.v1.autograph import experimental
  File "/home/hgandhi/.local/lib/python3.7/site-packages/tensorflow_core/_api/v1/autograph/experimental/__init__.py", line 10, in <module>
    from tensorflow.python.autograph.core.converter import Feature
  File "/home/hgandhi/.local/lib/python3.7/site-packages/tensorflow_core/python/autograph/core/converter.py", line 71, in <module>
    from tensorflow.python.autograph.pyct import cfg
  File "/home/hgandhi/.local/lib/python3.7/site-packages/tensorflow_core/python/autograph/pyct/cfg.py", line 41, in <module>
    from tensorflow.python.autograph.pyct import compiler
  File "/home/hgandhi/.local/lib/python3.7/site-packages/tensorflow_core/python/autograph/pyct/compiler.py", line 36, in <module>
    from tensorflow.python.autograph.pyct import origin_info
  File "/home/hgandhi/.local/lib/python3.7/site-packages/tensorflow_core/python/autograph/pyct/origin_info.py", line 31, in <module>
    from tensorflow.python.autograph.pyct import pretty_printer
  File "/home/hgandhi/.local/lib/python3.7/site-packages/tensorflow_core/python/autograph/pyct/pretty_printer.py", line 24, in <module>
    import termcolor
ModuleNotFoundError: No module named 'termcolor'
---------------------------------------------------------------------------
```

**Comment by RainierBarrett** [comment]
Finally able to get the notebooks to both run. Once you address my above comment on the header style (markdown cells), I'll approve this PR.

**Pull request merged**